services/dns/server/dns.go

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2022 Snowflake Inc. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+*/
+
+// Package server implements the sansshell 'Logging' service.
+package server
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/go-logr/logr"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/Snowflake-Labs/sansshell/services"
+	pb "github.com/Snowflake-Labs/sansshell/services/dns"
+)
+
+var (
+	// Create package level resolver such that it can be replaced during testing
+	defaultResolver = func() Resolver {
+		return resolver{}
+	}
+)
+
```

Comment:
This seems a bit overly complicated for a test fake.

You can just do:

var (
  resolver = net.DefaultResolver.LookupIP
)

and use it directly below and then in tests just replace that with a function that matches that signature. No need then for an interface and a struct, impl, etc.